### PR TITLE
Add CI to claim crates

### PR DESCRIPTION
**Pull request by Morganamilo**

### CODE REVIEW

File: .github/workflows/claim-crates.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+on:
+  push:
+    branches:
+      - master
+
+jobs:
+  claim-crates:
+    strategy:
+      matrix:
+        os: ["ubuntu-latest"]
```
Comment (coderobe) [inline .github/workflows/claim-crates.yml:12]: why use a matrix for this?

Comment (Morganamilo) [reply .github/workflows/claim-crates.yml:12]: Other workflows do so just staying consistent.

Comment (coderobe) [reply .github/workflows/claim-crates.yml:12]: hmm i don't think it makes sense here since this will never need to be ran on more than one

Comment (Morganamilo) [reply .github/workflows/claim-crates.yml:12]: Sure I'll remove it then. And from the other publish CI

**Review by alvicsam — APPROVED** [review]

File: .github/workflows/claim-crates.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    steps:
+      - uses: actions/checkout@8ade135a41bc03ea155e62e844d188df1ea18608 # v4.1.0
+
+      - name: Rust Cache
+        uses: Swatinem/rust-cache@3cf7f8cc28d1b4e7d01e3783be10a97d55d483c8 # v2.7.1
+        with:
+          cache-on-failure: true
+
+      - name: install parity-publish
+        run: cargo install parity-publish
```
Comment (altaua) [inline .github/workflows/claim-crates.yml:23]: can we pin the crate version?

Comment (Morganamilo) [reply .github/workflows/claim-crates.yml:23]: Any reason? It's a crate controlled by us so we push updates as needed. But I can pin it if that's wanted

Comment (Morganamilo) [reply .github/workflows/claim-crates.yml]: Pinned now.

**Review by coderobe — APPROVED** [review]

**Pull request merged**